sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -535,3 +537,77 @@ def _to_generated(self) -> GeneratedConfigurationSnapshot:
             retention_period=self.retention_period,
             tags=self.tags,
         )
+
+
+class ConfigurationSettingPropertiesPaged(PageIterator):
+    """An iterable of ConfigurationSetting properties."""
+
+    page_etag: str
+    """The etag of current page."""
+    
+    def __init__(self, command: Callable, **kwargs: Any) -> None:
+        super(ConfigurationSettingPropertiesPaged, self).__init__(
+            self._get_next_cb,
+            self._extract_data_cb,
+            continuation_token=kwargs.get("continuation_token"),
+        )
+        self._command = command
+        self._key = kwargs.get("key")
+        self._label = kwargs.get("label")
+        self._accept_datetime = kwargs.get("accept_datetime")
+        self._select = kwargs.get("select")
+        self._cls = kwargs.get("cls")
+
+    
+    def _get_next_cb(self, continuation_token):  # pylint: disable=inconsistent-return-statements
+        response = self._command(
+            key=self._key,
+            label=self._label,
+            accept_datetime=self._accept_datetime,
+            select=self._select,
+            cls=self._cls,
+        ).by_page(continuation_token=continuation_token)
+        next(response)
+        self.page_etag = response._response.http_response.headers['Etag']
```

Comment:
How do we use page_etag?

Reply:
We add the usage in tests instead of samples as we are not going to expose the monitor by page etag feature to broader customers: https://github.com/Azure/azure-sdk-for-python/pull/34346/files#diff-17ac75f747a78bee79417a869c8ce4115b634cd429f34d77b2b37431fcad255c

Reply:
@annatisch, should we simply expose a lower level `headers()` method to access the raw headers? This is what we have talked about for DPG models in general. As it stands, we are asymmetrical (we lift the per-page etag up as a typed property, but we are not accepting it as input in a typed API....)

Reply:
Also, tagging @iscai-msft since I suspect that she is interested in header-access-from-models.

Reply:
Yeah - simply exposing the response headers on the "Page" object is probably something we can do in Core. Might even be able to get that in before this GA's so we can remove the custom implementation in AppConfig.
That would also make more sense in case of being pinned to a prior API version where returned headers may vary.